src/dispersion_relation.jl

```diff
@@ -0,0 +1,119 @@
+@doc raw"""
+    LinearDispersionRelation(ref_height)
+
+A struct representing a linear dispersion relation ``\omega(k)`` of an equation. The reference
+water height `h0` is given by `ref_height`. A dispersion relation can be called as
```

Comment:
```suggestion
water height ``h_0`` is given by `ref_height`. A dispersion relation can be called as
```

Math mode like in the docstring of `wave_speed`?